src/Gui/DlgSettingsWorkbenchesImp.cpp

```diff
@@ -0,0 +1,576 @@
+
+    // Tokenize the comma-separated list
+    _backgroundAutoloadedModules.clear();
+    std::stringstream stream(autoloadCSV);
+    std::string workbench;
+    while (std::getline(stream, workbench, ','))
+        _backgroundAutoloadedModules.push_back(workbench);
+
+    buildWorkbenchList();
+
+    //We set the startup setting after building the list so that we can put only the enabled wb.
+    setStartWorkbenchComboItems();
+
+    {
+        QSignalBlocker sigblk(ui->CheckBox_WbByTab);
+        ui->CheckBox_WbByTab->onRestore();
+    }
+}
+
+/**
+Build the list of unloaded workbenches.
+*/
+void DlgSettingsWorkbenchesImp::buildWorkbenchList()
+{
+    QSignalBlocker sigblk(ui->wbList);
+
+    QStringList workbenches = Application::Instance->workbenches();
+    QStringList enabledWbs = getEnabledWorkbenches();
+    QStringList disabledWbs = getDisabledWorkbenches();
+
+    //First we add the enabled wbs in their saved order.
+    for (const auto& wbName : enabledWbs) {
+        if (workbenches.contains(wbName)) {
+            addWorkbench(wbName, true);
+        }
+        else {
+            Base::Console().Warning("Ignoring unknown %s workbench found in user preferences.", wbName.toStdString().c_str());
+        }
+    }
+    //Second we add workbench in alphabetical order that are either Disabled, or !enabled && !disabled, ie newly added wb.
+    for (const auto& wbName : workbenches) {
+        if (disabledWbs.contains(wbName)) {
+            addWorkbench(wbName, false);
+        }
+        else if (!enabledWbs.contains(wbName)) {
+            Base::Console().Warning("Adding unknown %s workbench.", wbName.toStdString().c_str());
+            addWorkbench(wbName, false);
+        }
+    }
+}
+
+void DlgSettingsWorkbenchesImp::addWorkbench(const QString& wbName, bool enabled)
+{
+    if (wbName.toStdString() == "NoneWorkbench")
+        return; // Do not list the default empty Workbench
+
+    bool isStartupWb = wbName.toStdString() == _startupModule;
+    bool autoLoad = std::find(_backgroundAutoloadedModules.begin(), _backgroundAutoloadedModules.end(),
+        wbName.toStdString()) != _backgroundAutoloadedModules.end();
+    wbListItem* widget = new wbListItem(wbName, enabled, isStartupWb, autoLoad, ui->wbList->count(), this);
+    connect(widget, &wbListItem::wbToggled, this, &DlgSettingsWorkbenchesImp::wbToggled);
+    auto wItem = new QListWidgetItem();
+    wItem->setSizeHint(widget->sizeHint());
+    ui->wbList->addItem(wItem);
+    ui->wbList->setItemWidget(wItem, widget);
+}
+
+
+QStringList DlgSettingsWorkbenchesImp::getEnabledWorkbenches()
+{
+    QString enabled_wbs;
+    QStringList enabled_wbs_list;
+    ParameterGrp::handle hGrp;
+    QString allWorkbenches = QString::fromLatin1("ALL");
+
+    hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Workbenches");
+    enabled_wbs = QString::fromStdString(hGrp->GetASCII("Enabled", allWorkbenches.toStdString().c_str()).c_str());
```

Comment:
The c_str() at the end is not needed and creates an unnecessary tmp. copy of std::string